File: src/henaxel/workbench/Toolbar.java
package henaxel.workbench;

import javax.swing.*;
import java.awt.*;


public class Toolbar extends JPanel {
    private int width;

    public Toolbar(String name, int width, Tool... tools) {
        setName(name);
        this.width = width;
        addTools(tools);

    }

    public void addTools(Tool... tools) {
        for(Tool tool : tools) {
            add(tool);
        }
    }

    public void setAnchor(String anchor) {
        switch(anchor) {
            case BorderLayout.NORTH: case BorderLayout.SOUTH:
                setLayout(new BoxLayout(this, BoxLayout.LINE_AXIS));
                break;
            case BorderLayout.WEST: case BorderLayout.EAST:
                setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
                break;
        }
    }

}

File: src/henaxel/workbench/Toolbar.kt
package henaxel.workbench

import java.awt.BorderLayout
import javax.swing.BoxLayout
import javax.swing.JPanel

class Toolbar(name: String, width: Int, vararg tools: Tool) : JPanel() {


    init {
        setName(name)
        addTools(tools)
    }

    fun addTools(tools: Array<out Tool>) {
        for(tool in tools) add(tool)
    }

    fun setAnchor(anchor: String) {
        when(anchor) {
            BorderLayout.NORTH  -> layout = BoxLayout(this, BoxLayout.LINE_AXIS)
            BorderLayout.SOUTH  -> layout = BoxLayout(this, BoxLayout.LINE_AXIS)
            BorderLayout.EAST   -> layout = BoxLayout(this, BoxLayout.PAGE_AXIS)
            BorderLayout.WEST   -> layout = BoxLayout(this, BoxLayout.PAGE_AXIS)
        }
    }
}